cmd/argocd/commands/login.go

```diff
@@ -38,35 +47,48 @@ func NewLoginCommand(globalClientOpts *argocdclient.ClientOptions) *cobra.Comman
 			errors.CheckError(err)
 			if !tlsTestResult.TLS {
 				if !globalClientOpts.PlainText {
-					askToProceed("WARNING: server is not configured with TLS. Proceed (y/n)? ")
+					cli.AskToProceed("WARNING: server is not configured with TLS. Proceed (y/n)? ")
 					globalClientOpts.PlainText = true
 				}
 			} else if tlsTestResult.InsecureErr != nil {
 				if !globalClientOpts.Insecure {
-					askToProceed(fmt.Sprintf("WARNING: server certificate had error: %s. Proceed insecurely (y/n)? ", tlsTestResult.InsecureErr))
+					cli.AskToProceed(fmt.Sprintf("WARNING: server certificate had error: %s. Proceed insecurely (y/n)? ", tlsTestResult.InsecureErr))
 					globalClientOpts.Insecure = true
 				}
 			}
-
-			username, password = cli.PromptCredentials(username, password)
-
 			clientOpts := argocdclient.ClientOptions{
 				ConfigPath: "",
 				ServerAddr: server,
 				Insecure:   globalClientOpts.Insecure,
 				PlainText:  globalClientOpts.PlainText,
 			}
-			conn, sessionIf := argocdclient.NewClientOrDie(&clientOpts).NewSessionClientOrDie()
-			defer util.Close(conn)
+			acdClient := argocdclient.NewClientOrDie(&clientOpts)
+			setConn, setIf := acdClient.NewSettingsClientOrDie()
+			defer util.Close(setConn)
+
+			acdSet, err := setIf.Get(context.Background(), &settings.SettingsQuery{})
+			errors.CheckError(err)
+
+			ctxName = cli.PromptMessage("Enter a name for this context", ctxName)
+			username = cli.PromptUsername(username)
 
-			sessionRequest := session.SessionCreateRequest{
-				Username: username,
-				Password: password,
+			// Perform the login
+			var tokenString string
+			if username == "admin" || !ssoConfigured(acdSet) {
```

Comment:
I think what I have is not ideal and I'm open to a better way. The problem is there are two different types of logins: SSO and local user. Right now `argo login` tries to infer based on server settings and the username. The assumption is that there are not really any local users other than `admin`, which I think will always be true. I don't see any point in implementing any sort of user management. Anyone running a CD solution for kubernetes, should have some type of IDp.

My current problem with what I have, is that username is meaningless with respect to SSO login. That value is essentially ignored and is only used to decide the code path of local login vs. sso login. So instead of checking username, I'm thinking of having either a --sso flag or a --local flag. WDYT?